### PR TITLE
fix restart function for AWS instances

### PR DESCRIPTION
**Pull request by roydahan**
When restarting a scylla node with ephemeral drive like i3/i2, we need to remove the entry from /etc/fstab so the ami_setup will succeed.
Also, need to add replace_address to scylla.yaml (only if not already exist, because the same nemesis can run on the same host several times).

### CODE REVIEW

**Comment by roydahan** [comment]
Testing it now, but meanwhile you can review to find any issue.
I'll need it in 3.0 ASAP.

**Comment by amoskong** [comment]
scylla-ami-setup is necessary because the RAID needs to be resetup.

The ec2 user data needs to be updated:
- pass the `--seeds` parameter
- remove `--stop-services` parameter
- maybe update `--bootstrap` to true?

**Comment by amoskong** [comment]
from: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

```
If you stop an instance, modify its user data, and start the instance, the updated user data is not executed when you start the instance.
```

**Comment by amoskong** [comment]
Why don't we just create devices & setup raid by executing `/usr/lib/scylla/scylla-ami/scylla_create_devices`? We don't need to run scylla-ami-setup.

**Comment by roydahan** [comment]
You was the one who suggested just to delete the ami_configured file and restart the node.
I don’t know what should be the procedure in this case, I think it should be like replace node best practice.

**Comment by amoskong** [comment]
I didn't find interface in boto3 lib to update user-data, it might not work
after modify (
https://github.com/scylladb/scylla-cluster-tests/pull/679#issuecomment-436392239
)

However, I tried to only create devices & setup raid by executing
/usr/lib/scylla/scylla-ami/scylla_create_devices, then we don't need to
rerun full scylla-ami-setup service. It runs well in builder4. I will send
another PR for review.

**Comment by bentsi** [comment]
@amoskong Actually you can modify UserData attribute :)
You can read about it here:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.modify_instance_attribute

Note:

> UserData (dict) --
> Changes the instance's user data to the specified value. If you are using an AWS SDK or command line tool, base64-encoding is performed for you, and you can load the text from a file. Otherwise, you must provide base64-encoded text.
> 
> Value (bytes) --

But any way I think that running `/usr/lib/scylla/scylla-ami/scylla_create_devices` is better solution.

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -329,7 +330,16 @@ def restart(self):
         # When using storage optimized instances like i2 or i3, the data on disk is deleted upon STOP. therefore, we
         # need to setup the instance and treat it as a new instance.
         if any(ss in self._instance.instance_type for ss in ['i3', 'i2']):
-            self.remoter.run('sudo rm -f /etc/scylla/ami_configured')
+            clean_script = dedent("""
+                sudo rm -f /etc/scylla/ami_configured
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: remove `sudo`
the whole script is run as `sudo`

File: sdcm/cluster_aws.py [hunk 2]
```diff
@@ -329,7 +330,16 @@ def restart(self):
...
+                sudo sed -e '/.*scylla/ s/^/# /g' -i /etc/fstab
+                sudo sed -e 's/replace_address_on_first_boot/#replace_address_on_first_boot/g' -i /etc/scylla/scylla.yaml
+            """)
+            self.remoter.run("sudo bash -cxe '%s'" % clean_script)
+            output = self.remoter.run('sudo grep replace_address: /etc/scylla/scylla.yaml', ignore_status=True)
+            if 'replace_address:' not in output.stdout:
```
Comment (bentsi) [inline sdcm/cluster_aws.py]: this won't work if we will run `restart()` method more than one time and the on each of the restarts we get 
different IP

Comment (roydahan) [reply sdcm/cluster_aws.py]: This is the internal IP address and it should remain the same IP.

**Comment by roydahan** [comment]
This PR will probably not be merged at all, we found it it's still not working even after all these fixes.
@amoskong will send a new PR that should not delete the ami_configured and just rebuild raid.

The main issue here is that we don't really have a procedure what to do in that case.

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -329,7 +330,16 @@ def restart(self):
         # When using storage optimized instances like i2 or i3, the data on disk is deleted upon STOP. therefore, we
         # need to setup the instance and treat it as a new instance.
         if any(ss in self._instance.instance_type for ss in ['i3', 'i2']):
-            self.remoter.run('sudo rm -f /etc/scylla/ami_configured')
+            clean_script = dedent("""
+                sudo sed -e '/.*scylla/s/^/#/g' -i /etc/fstab
+                sudo sed -e 's/replace_address_on_first_boot/#replace_address_on_first_boot/g' -i /etc/scylla/scylla.yaml
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: @amoskong, in this case, I don't think we need this line for "replace_address".

File: sdcm/cluster_aws.py [hunk 4]
```diff
@@ -339,6 +349,11 @@ def restart(self):
                        self._instance.public_ip_address)
         self.remoter.hostname = self._instance.public_ip_address
 
+        if any(ss in self._instance.instance_type for ss in ['i3', 'i2']):
+            self.remoter.run('sudo /usr/lib/scylla/scylla-ami/scylla_create_devices')
```
Comment (roydahan) [inline sdcm/cluster_aws.py]: Is this script add a record to /etc/fstab?

Comment (amoskong) [reply sdcm/cluster_aws.py]: scylla_create_devices script will run scylla_raid_setup with parameter `--update-fstab`
The /etc/fstab file will be updated.

**Pull request merged**